src/test/java/com/powsybl/openloadflow/network/ShuntNetworkFactory.java

```diff
@@ -125,4 +125,47 @@ public static Network createWithTwoShuntCompensators() {
                 .add();
         return network;
     }
+
+    public static Network createWithGeneratorAndShunt() {
+        Network network = create();
+        VoltageLevel vl3 = network.getVoltageLevel("vl3");
+        vl3.newGenerator()
+                .setId("g2")
+                .setConnectableBus("b3")
+                .setBus("b3")
+                .setTargetP(0)
+                .setTargetV(393)
+                .setMinP(0)
+                .setMaxP(150)
+                .setVoltageRegulatorOn(true)
+                .add();
+        return network;
+    }
+
+    public static Network createWithGeneratorAndShuntNonImpedant() {
```

Comment:
Not really related to this PR, but this network is causing the following ERROR log to be issued:
`ERROR c.p.o.n.i.LfNetworkLoaderImpl | Controller shunt vl3_0_controller_shunt_compensators is already in a shunt voltage control. The second controlled bus vl3_0 is ignored`
Which is imo a false positive that should be fixed.

Reply:
"imo"? Before the security analysis where we create 2 shunt voltage controls, a MAIN one and a DEPENDENT one, we run 2 load flows. In these two load flows, a single shunt voltage control with controlled bus b3 is created. This control has two shunt references in it.